ivy/functional/backends/torch/general.py

```diff
@@ -133,12 +133,20 @@ def to_numpy(
         x = x.resolve_neg().resolve_conj()
         if copy:
             if x.dtype is torch.bfloat16:
-                default_dtype = ivy.default_float_dtype(as_native=True)
-                if default_dtype is torch.bfloat16:
-                    x = x.to(torch.float32)
-                else:
-                    x = x.to(default_dtype)
-                return x.detach().cpu().numpy().astype("bfloat16")
+                try:
+                    return x.detach().cpu().numpy().astype("bfloat16")
+                except TypeError:
+                    # although ml_dtypes would enable bfloat16
+                    # behaviour in numpy, but torch is restricting
+                    # it here, so we need to do this
+                    default_dtype = ivy.default_float_dtype(as_native=True)
+                    if default_dtype is torch.bfloat16:
+                        x = x.to(torch.float32)
+                    else:
+                        x = x.to(default_dtype)
+                    from ml_dtypes import bfloat16
```

Comment:
The `import ml_dtypes` statement should be done in the `ivy/functional/ivy/general.py` file above the `import numpy as np` import. We wouldn't need to import it again here as a result.
Secondly, I think even without the `try-except` we can just have the logic from line `142` to `149` directly under the `if` block on line `135` and that should work as well.
Also, let's create an issue on the `torch` repo and link it here with a `ToDo` comment.
Thanks @RickSanchezStoic 😄